tests/track/test_auth_wrapper.py

```diff
@@ -0,0 +1,183 @@
+import os
+from unittest.mock import MagicMock, patch
+
+from track.auth_wrapper import create_wrapper_app
+
+
+@patch.dict(
+    os.environ,
+    {
+        'GOOGLE_CLIENT_ID': 'test_client_id',
+        'GOOGLE_CLIENT_SECRET': 'test_client_secret',
+        'ALLOWED_EMAIL': 'user@example.com,admin@example.com',
+    },
+)
+@patch('track.auth_wrapper.get_session_secret')
+@patch('track.auth_wrapper.get_internal_api_key')
+def test_create_wrapper_app(mock_get_api_key, mock_get_session_secret):
+    """Test that create_wrapper_app creates a FastAPI app with proper configuration."""
+    mock_get_api_key.return_value = 'test_api_key_123'
+    mock_get_session_secret.return_value = 'test_session_secret_456'
+
+    app, internal_api_key = create_wrapper_app('test_app')
+
+    assert app is not None
+    # FastAPI uses 'title' differently than name, let's check what we can verify
+    assert hasattr(app, 'title')  # App has a title attribute
+    assert internal_api_key == 'test_api_key_123'
+
+    mock_get_api_key.assert_called_once_with('test_app')
+    mock_get_session_secret.assert_called_once_with('test_app')
+
+
+@patch.dict(
+    os.environ,
+    {
+        'GOOGLE_CLIENT_ID': 'test_client_id',
+        'GOOGLE_CLIENT_SECRET': 'test_client_secret',
+        'ALLOWED_EMAIL': 'user@example.com',
+    },
+)
+@patch('track.auth_wrapper.get_session_secret')
+@patch('track.auth_wrapper.get_internal_api_key')
+def test_service_authorization(mock_get_api_key, mock_get_session_secret):
+    """Test service authorization using API key."""
+    mock_get_api_key.return_value = 'service_api_key'
+    mock_get_session_secret.return_value = 'session_secret'
+
+    app, _ = create_wrapper_app('test_app')
+
+    # Test the authorization logic directly since accessing middleware is complex
```

Comment:
I see what you mean. But in that case, I think we should either refactor it to be testable or just not test it.